packages/axelar-soroban-std/src/shared_interfaces.rs

```diff
@@ -105,35 +102,23 @@ fn complete_migration(env: &Env) {
         .remove(&storage::DataKey::SharedInterfaces_Migrating);
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 pub struct UpgradedEvent {
     version: String,
 }
 
-impl UpgradedEvent {
-    pub fn topic() -> impl Topics {
+impl Event for UpgradedEvent {
+    fn topics(&self) -> impl Topics + Debug {
         (symbol_short!("upgraded"),)
     }
 
-    pub fn data(&self) -> impl IntoVal<Env, Val> {
-        (self.version.to_val(),)
+    fn data(&self) -> impl IntoVal<Env, Val> + Debug {
+        (self.version.clone(),)
```

Comment:
not efficient to have to create a copy of the field to create the topic and data. Should we consider combining the two into something like `fn topic_and_data(self) -> (impl Topics + Debug, impl IntoVal<Env, Val> + Debug)` so it can take `self` instead of a reference

Reply:
that clone slipped in from some experimentation. The data has to be converted into a val anyway, so there isn't a good reason to muddle up the interface